components/common/infinite_scroll.jsx

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const SCROLL_BUFFER = 10;
+
+export default class InfiniteScroll extends React.PureComponent {
+    static defaultProps = {
+        bufferValue: SCROLL_BUFFER,
+    };
+
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+
+        /**
+         * Function that is called to load more items
+         */
+        callBack: PropTypes.func.isRequired,
+
+        /**
+         * Boolean to indicate that no more data will be loaded
+         */
+        endOfData: PropTypes.bool.isRequired,
+
+        /**
+         * Message to display when all the data has been scrolled through
+         */
+        endOfDataMessage: PropTypes.string.isRequired,
```

Comment:
We usually don't have end of data messages so can we keep this optional.

Reply:
Alright... let me do that.

Reply:
@sudheerDev I made this optional with the default being an empty string.